### PR TITLE
feat: history functionality

### PR DESCRIPTION
**Pull request by pankaj-prog**

### CODE REVIEW

File: src/context/HistoryContext.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+const HistoryProvider = ({ children }) => {
+  const [historyVideos, setHistoryVideos] = useState([]);
+  const { encodedToken } = useAuth();
+
+  const { makeRequest: addToHistoryRequest, response: addToHistoryResponse } =
+    useAxios();
+
+  const {
+    makeRequest: removeFromHistoryRequest,
```
Comment (bhtibrewal) [inline src/context/HistoryContext.jsx:17]: Appreciation: good thought here

**Review by bhtibrewal — APPROVED** [review]
Good work on the code; clean and understandable, you can go ahead and merge

**Review by SinchanaMS — APPROVED** [review]
The code is well-written and neat! Didn't find any issues, good work!

**Review by talrejah02 — APPROVED** [review]
Clean code, well structured.
Good implementation of history.

**Pull request merged**